app/src/main.cpp

```diff
@@ -219,6 +219,25 @@ namespace scitt
         throw BadRequestError(errors::InvalidInput, e.what());
       }
 
+      // Retrieve current enclave measurement of this node
+      // See ccf logic in `/quotes/self`
+      std::string measurement;
+      auto nodes = ctx.tx.ro<ccf::Nodes>(ccf::Tables::NODES);
+      auto node_info = nodes->get(context.get_node_id());
+      if (node_info.has_value() && node_info->code_digest.has_value())
+      {
+        measurement = node_info->code_digest.value();
+      }
+      else
+      {
+#if defined(INSIDE_ENCLAVE) && !defined(VIRTUAL_ENCLAVE)
+        // Node should always get a valid cached measurement on startup
+        throw InternalError("Unexpected state - node has no code id");
+#else
+        measurement = "";
```

Comment:
Why set it to an empty string if it could not be retrieved when running in virtual platform? IIUC, it would result in a different mrenclave value in the receipt, as it's not the same as a 64-0s string, making it potentially inconsistent.

Reply:
updated to the expected one